java/src/org/openqa/selenium/os/ExternalProcess.java

```diff
@@ -264,7 +264,7 @@ public boolean waitFor(Duration duration) throws InterruptedException {
     boolean exited = process.waitFor(duration.toMillis(), TimeUnit.MILLISECONDS);
 
     if (exited) {
-      worker.join();
+      worker.join(30000);
```

Comment:
How about `thread.interrupt()` after the join, this might stop the hanging thread.

Reply:
If it hangs on join() will it ever get to interrupt()?

Reply:
The timeout should be added to the join call too.
```
worker.join(30000);
worker.interrupt();
```

Reply:
is 30 seconds the right timeout?